README.md

```diff
@@ -1,2 +1,199 @@
+
+```kotlin
+class LoginActivity : AppCompatActivity(), LoginView {
+    override fun bind(newState: LoginState) { ... }
+}
+```
+
+### RxStore
+
+It receives and manages all the possible states of a `View` so you don't need to worry, because we take care about statefulness and performance.
+
+To create one you just need to pass an `State` class as parameter and define the related initial state:
+
+```kotlin
+RxStore(LoginState(StateType.Initial))
+```
+
+> `RxStore` is an implementation of `Store` that works with Reactive Extensions
+
+The stream created by it can be handled inside the `View` implementation.
+
+```kotlin
+override fun onCreate(savedInstanceState: Bundle?) {
+    super.onCreate(savedInstanceState)
+    setContentView(R.layout.activity_login)
+
+    store.stateChanges()
+         .observeOn(AndroidSchedulers.mainThread())
+         .subscribe { bind(it) }
+}
+
+override fun bind(newState: LoginState) {
+   when (newState.type) {
+     StateType.Success -> success()
+     StateType.Error -> error()
+     StateType.Loading -> loading()
+   }
+}
+```
+
+
+### RxJob
+
+Small units of work separated from the other layers and that warns the `RxStore` layer when the job is finished successfully or not.
+
+Just need to inherit your use case class from `RxJob` (specifying the kind of return of this job) and override the `run(input: T)` method with the properly work.  As it is returns a `Completable` you should handle the possible returns on the `RxStore` subscription, with `doOnSubscribe`, `doOnSuccess` and `doOnError` as shown in the next example:
+
+```kotlin
+class DoGoogleLoginJob @Inject constructor(
+    private val store: RxStore<LoginState>,
+    private val firebase: GoogleFirebaseAuth) : RxJob<Intent>() {
+
+  override fun bind(input: Intent): Completable {
+    return firebase.firebaseAuthWith(intent = input)
+        .doOnSubscribe { store.update { setType(StateType.Loading) } }
+        .doOnSuccess { store.update { setType(StateType.Success) } }
+        .doOnError { store.update { setType(StateType.Error) } }
+        .toCompletable()
+        .onErrorComplete()
+  }
+}
+```
+
+> `RxJob` is an implementation of `Job` that works with Reactive Extensions
+
+> In this example we are injecting the local parameters `store` and `firebase` with [dagger](https://github.com/google/dagger) we strong recommend that.
+
+### RxHub
+
+The hub is maybe the most important and disruptive layer of this framework. It is responsible to connect the `View` and the `Job`'s layers through the operator `bind(job: RxJob<T>)`(that's why all of your `View`'s methods needs to return an `Observable`).
+
+You can combine and/or filter actions before perform the `Job` itself. It acts kind like a bridge binding user actions with the use cases. 
+
+Just need to inherit the class `RxHub<T>`(specifying the `View` it will connect with) and override the `connect(view: T)` method with the binds to `RxJobs` you need.
```

Comment:
It's nice to mention that we have already implemented the disconnect method to release resources by disposing all the bound jobs.